nameko/containers.py

```diff
@@ -171,8 +170,8 @@ def __init__(self, service_cls, config, worker_ctx_cls=None):
         self.started = False
         self._worker_pool = GreenPool(size=self.max_workers)
 
-        self._active_threads = {}
-        self._protected_threads = set()
+        self._worker_threads = WeakKeyDictionary()
```

Comment:
OK, will remove. I completely agree on removing `container` from the `WorkerContext`. Bound extensions all have access to it anyway, we don't need to pass the reference around everywhere.